test/integration/builder/transaction.js

```diff
@@ -277,6 +279,41 @@ module.exports = function(knex) {
 
     });
 
+    it('#1040, #1171 - When pool is filled with transaction connections, Non-transaction queries should not hang the application, but instead throw a timeout error', function(done) {
+      this.timeout(15000);
+      //To make this test easier, I'm changing the pool settings to max 1.
+      var knexConfig = _.clone(knex.client.config);
+      knexConfig.pool.min = 0;
+      knexConfig.pool.max = 1;
+      knexConfig.acquireConnectionTimeout = 10000;
+
+      var knexDb = new Knex(knexConfig);
+
+      //Create a transaction that will occupy the only available connection, and avoid trx.commit.
+     knexDb.transaction(function(trx) {
+        trx.raw('SELECT 1 = 1').then(function() {
+
+          //No connection is available, so try issuing a query without transaction.
+          //Since there is no available connection, it should throw a timeout error based on `aquireConnectionTimeout` from the knex config.
+          knexDb.raw('select * FROM accounts WHERE username = ?', ['Test'])
+              .then(function(res) {
+                expect(res).to.not.exist();
```

Comment:
Is intention of this to always fail if this then block is executed? In that case `expect(false).to.be.ok()` would be more readable to me, since it fail independently from `res`.